submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/MongoDbConfig.kt

```diff
@@ -55,70 +37,30 @@ const val CHANGE_LOG_LOCK = "submitter_mongockLock"
         springContext: ApplicationContext,
-        mongoTemplate: MongoTemplate
+        mongoTemplate: MongoTemplate,
     ): ApplicationRunner {
         return createMongockConfig(mongoTemplate, springContext, CHANGE_LOG_CLASSES)
     }
 
     @Bean
-    override fun mongoClient(): MongoClient {
+    fun mongoClient(): MongoClient {
         return MongoClients.create(mongoUri)
     }
 
     @Bean
-    override fun customConversions(): MongoCustomConversions {
-        val converters = mutableListOf<Converter<*, *>>()
-        converters.add(docSubmissionConverter())
-        converters.add(submissionConverter())
-        converters.add(FileConverter(AttributeConverter()))
-        converters.add(DocFileConverter(DocAttributeConverter()))
-        converters.add(FileListDocFileConverter(FileConverter(AttributeConverter())))
-        converters.add(DocFileListDocFileConverter(DocFileConverter(DocAttributeConverter())))
-        return MongoCustomConversions(converters)
-    }
-
-    private fun docSubmissionConverter(): DocSubmissionConverter {
-        val docAttributeConverter = DocAttributeConverter()
-        val docFileConverter = DocFileConverter(docAttributeConverter)
-        val docFileListConverter = DocFileListConverter(docFileConverter)
-        val docFileTableConverter = DocFileTableConverter(docFileConverter)
-        val docLinkConverter = DocLinkConverter(docAttributeConverter)
-        val docLinksTableConverter = DocLinkTableConverter(docLinkConverter)
-        val docSectionConverter = DocSectionConverter(
-            docAttributeConverter,
-            docLinkConverter,
-            docLinksTableConverter,
-            docFileConverter,
-            docFileTableConverter,
-            docFileListConverter
-        )
-        return DocSubmissionConverter(docFileConverter, docSectionConverter, docAttributeConverter)
+    fun mongoTemplate(databaseFactory: MongoDatabaseFactory, converter: MappingMongoConverter): MongoTemplate {
+        return MongoTemplate(databaseFactory, converter)
     }
 
-    private fun submissionConverter(): SubmissionConverter {
-        val attributeConverter = AttributeConverter()
-        val fileConverter = FileConverter(attributeConverter)
-        val fileListConverter = FileListConverter(fileConverter)
-        val fileTableConverter = FileTableConverter(fileConverter)
-        val linkConverter = LinkConverter(attributeConverter)
-        val linksTableConverter = LinkTableConverter(linkConverter)
-        val sectionConverter = SectionConverter(
-            attributeConverter,
-            linkConverter,
-            linksTableConverter,
-            fileConverter,
-            fileTableConverter,
-            fileListConverter
-        )
-        return SubmissionConverter(sectionConverter, attributeConverter, fileConverter)
+    @Bean
+    fun mongoDbFactory(): MongoDatabaseFactory {
+        return SimpleMongoClientDatabaseFactory(mongoClient(), mongoDatabase)
```

Comment:
I'm curious about why are you using instantiation here instead of the parameter notation as in

```
@Bean
fun mongoDbFactory(
    mongoClient: MongoClient,
): MongoDatabaseFactory {
    return SimpleMongoClientDatabaseFactory(mongoClient, mongoDatabase)
}
```